core/stateless/witness.go

```diff
@@ -38,9 +38,9 @@ type HeaderReader interface {
 type Witness struct {
 	context *types.Header // Header to which this witness belongs to, with rootHash and receiptHash zeroed out
 
-	Headers []*types.Header     // Past headers in reverse order (0=parent, 1=parent's-parent, etc). First *must* be set.
-	Codes   map[string]struct{} // Set of bytecodes ran or accessed
-	State   map[string]struct{} // Set of MPT state trie nodes (account and storage together)
+	Headers []*types.Header `json:"headers"` // Past headers in reverse order (0=parent, 1=parent's-parent, etc). First *must* be set.
+	Codes   []hexutil.Bytes `json:"codes"`   // Set of bytecodes ran or accessed
+	State   []hexutil.Bytes `json:"state"`   // Set of MPT state trie nodes (account and storage together)
```

Comment:
that doesn't look right, checking for a code or a slot would require a linear search each time.